### ISSUE TITLE
Require help in implementing drag and drop for Grid View

### ISSUE THREAD

**Issue by vinayak-garg** [h6ah4i/android-advancedrecyclerview#86]
I have made some progress in implementing grid view for this library. You can take a look at my fork here - https://github.com/vinayak-garg/android-advancedrecyclerview/tree/grid_layout

I am not able to understand the library code completely, so I need some help to implement grid view support.

Take a look at this screen capture video to see major issues that I need to resolve - https://www.youtube.com/watch?v=6EkTQgv4izE

Notice 3 major issues-
1) While dragging item 0 to 4, midway the 'move' operation is triggered. After that, the dragging item simply jumps
2) On long pressing item 6, position for item 1 and item 4 become blank. Also the draggable item is not under the finger.
3) While animating diagonal movement of item 4 to item 3 the item 4 is hidden by item 5, 6.

I made some simplifications in the library code to suit my requirements. However if I am able to get it working correctly I will fix them. 

Also note that my target is to support grid layout only for drag and drop. I don't need swiping and expandable view in grid layout.

Major changes that I have made so far to support grid view-
1) Change int/float values to Point wherever 2 dimensional values are required
2) Add logic to detect 1 of the 8 direction in which user is going
3) Select an array of items to swap (instead of just one)

Any help is highly appreciated :) If you need help with my changes, please ask.

**Comment by vinayak-garg** [comment]
So I managed to get the drag and drop in grid view working. :smiley:  It is not as smooth as the original Vertical layout implementation of this library. But it is smooth enough for most use cases. You can check out this branch - https://github.com/vinayak-garg/android-advancedrecyclerview/tree/grid_layout
In current form I need to do some clean ups. Like I have hard coded the column count as 4. And some of the things are not general enough, like accounting for all possible margins and padding. But it is not that hard to customise for your needs.

**Comment by h6ah4i** [comment]
Oh, you beat me to it! I have just uploaded my experimental GridLayoutManager support to this repository. It's not based on your code, but I got an idea from your old implementation. I will check your latest code later and it will be included in next major update.

https://github.com/h6ah4i/android-advancedrecyclerview/tree/feature/support_grid_layout_manager

**Comment by vinayak-garg** [comment]
Awesome! so I wouldn't have to clean up my code. ;) 

The main issue I faced while adding grid layout support, was how to get the `SwapTargetItemOperator` working correctly. Once I decided to skip it altogether, I was able to proceed. I lost some smoothness, but the result was still great.

I tried out 3 libraries for grid view drag and drop. None of them worked. Your library was the only one with which I could get good experience. Thanks a ton buddy!
